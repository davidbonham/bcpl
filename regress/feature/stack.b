GET "libhdr"

GLOBAL $( a: 128; b: 129; fin_l: 200; fin_p: 201 $)

LET foo(p, l) BE $(
    writes("in foo*n")
    longjump(fin_p, fin_l)
$)

LET start() BE $(
    a := 10
    b := 20
    fin_p, fin_l := level(), error
    foo(a, b)
    writes("normal return from foo*n")
error:
    writes("correctly arriving at error*n")
$)
