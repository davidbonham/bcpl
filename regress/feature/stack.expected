-- stdout -------------------------------------------
in foo
correctly arriving at error

-- stderr -------------------------------------------

